src/SettingsSystem.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <memory>
+#include <yaml-cpp/yaml.h>
+
+#include "EventSystem.h"
+
+struct InputSettings;
+struct AudioSettings;
+
+class SettingsSystem : public EventSystem<SettingsSystem> {
+public:
+    SettingsSystem(std::string config_file);
+    bool reload_settings();
+
+    std::shared_ptr<InputSettings> get_input_settings();
+    std::shared_ptr<AudioSettings> get_audio_settings();
```

Comment:
Just return a const reference instead of a shared_ptr.

And they should be const functions.